api4/config.go

```diff
@@ -157,9 +157,10 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 		*cfg.PluginSettings.MarketplaceURL = *appCfg.PluginSettings.MarketplaceURL
 	}
 
-	if cfg.PluginSettings.PluginStates[model.PluginIdFocalboard].Enable && cfg.FeatureFlags.BoardsProduct {
-		c.Err = model.NewAppError("EnablePlugin", "app.plugin.product_mode.app_error", map[string]any{"Name": model.PluginIdFocalboard}, "", http.StatusInternalServerError)
-		return
+	// Do not allow Focalboard plugin to be enabled if Boards
+	// is running as a product.
+	if cfg.FeatureFlags.BoardsProduct {
+		cfg.PluginSettings.PluginStates[model.PluginIdFocalboard].Enable = false
```

Comment:
To clarify, when we try to enable boards plugin in product mode we are don't want to return an error right. I still think this is an error case because the configuration is not consistent. The problem here is that this setting shouldn't be `true`. My proposal would be logging an warning message here to indicate that "config is wrong and we are ignoring that". WDYT?

Reply:
Actually, I do get the error message in system console on enabling Focalboard plugin. So I'm adding the warning message here now.

Reply:
Updated. Please take a look again.

Reply:
Thanks, yes if we are trying to enable boards plugin in product mode it should be an error case and admin should be informed somehow.